### ISSUE TITLE
Arm backend: Quantized Arm `sigmoid` TABLE generation uses an invalid int8 input code sequence when `qmin=-127`

### ISSUE THREAD

**Issue by Rob-Hughes-Arm** [pytorch/executorch#18873]
### 🐛 Describe the bug

# Bug: quantized Arm `sigmoid` TABLE generation uses an invalid int8 input code sequence when `qmin=-127`

## Summary

Quantized `aten.sigmoid.default` on the Arm backend is lowered to TOSA `TABLE`, but the generated int8 lookup table is wrong when `InsertTableOpsPass.generate_8bit_table_values()` is invoked with:

- `qmin = -127`
- `qmax = 127`
- `dtype = torch.int8`

The problem is in `executorch/backends/arm/_passes/insert_table_ops.py`, in `InsertTableOpsPass.generate_8bit_table_values()`:

```python
torch.linspace(
    start=in_quantargs.qmin,
    end=in_quantargs.qmax,
    steps=256,
    dtype=torch.int8,
)
```

For `qmin=-127`, `qmax=127`, `steps=256`, this does **not** produce a valid 256-entry integer code sequence. That range contains only **255 distinct int8 codes**, so `torch.linspace(..., dtype=torch.int8)` ends up duplicating `0` and shifting part of the table by one code.

This causes quantized sigmoid outputs to disagree with the PT2E q/dq eager reference when the sigmoid node reaches this pass with those input qparams.

This report is intentionally scoped to the `qmin=-127`, `qmax=127`, `dtype=torch.int8` case. It is **not** claiming that every int8 Arm sigmoid export is broken.

## Expected behavior

For a 256-entry TOSA int8 table, the table input domain should cover the full int8 code range exactly once:

```text
[-128, -127, -126, ..., 126, 127]
```

Using `torch.linspace(start=qmin, end=qmax, steps=256, dtype=torch.int8)` is not a correct way to construct that domain when `qmin..qmax` does not already contain 256 distinct integer codes.

If the implementation intentionally wants to derive the table from the quantized input domain instead of the full TOSA int8 domain, it still needs an explicit integer mapping with no duplicates or off-by-one shifts.

For quantized sigmoid specifically, the emitted LUT should match the q/dq eager reference:

```text
Q_in -> DQ_in -> sigmoid -> Q_out
```

## Actual behavior

The current code generates the wrong table input codes around the zero crossing and therefore produces the wrong LUT values for part of the negative range.

### Minimal pure-Python repro

```python
import torch

x = torch.linspace(start=-127, end=127, steps=256, dtype=torch.int8)

print("unique count:", torch.unique(x).numel())
print("indices 120..136:")
for i in range(120, 137):
    print(i, int(x[i]))
```

### Actual output

```text
unique count: 255

indices 120..136:
120 -7
121 -6
122 -5
123 -4
124 -3
125 -2
126 -1
127 0
128 0
129 1
130 2
131 3
132 4
133 5
134 6
135 7
136 8
```

### Expected output

```text
unique count: 256

indices 120..136:
120 -8
121 -7
122 -6
123 -5
124 -4
125 -3
126 -2
127 -1
128 0
129 1
130 2
131 3
132 4
133 5
134 6
135 7
136 8
```

## Why this matters

Because the table entries are generated from the wrong input codes, the emitted sigmoid table is effectively shifted by one code through the affected region.

Example from a real quantized sigmoid export:

- input qparams:
  - `scale = 0.15988604724407196`
  - `zero_point = -17`
- output qparams:
  - `scale = 0.0039350856095552444`
  - `zero_point = -127`

For input code `q_in = -17`, the eager q/dq reference expects:

```text
dequant(-17) = 0.0
sigmoid(0.0) = 0.5
quantize(0.5) = 0
```

But the emitted table entry at that index is generated from code `-16`, so the table output becomes:

```text
table[-17] == quantize(sigmoid(dequant(-16))) == 10
```

That is exactly the bad output observed in the backend result.

## End-to-end reproducible

This tiny quantized sigmoid export reproduces the issue for the affected `qmin=-127` case.

### Model

```python
import torch
import torch.nn as nn

class TinySigmoidProbe(nn.Module):
    def forward(self, x: torch.Tensor) -> torch.Tensor:
        return torch.sigmoid(x)
```

### Input

```python
x = torch.tensor(
    [[[[ -17.596058, -6.3985662, -1.4396775, -0.31992832,
          0.0, 0.15996416, 1.1197491, 3.5192115, 23.03484 ]]]],
    dtype=torch.float32,
)
```

### Observed PT2E q/dq eager output

```text
[
  0.0,
  0.0,
  0.19281919,
  0.42105415,
  0.49975586,
  0.53910673,
  0.75553644,
  0.97196615,
  0.99951172,
]
```

### Observed backend/scenario output

```text
[
  0.0,
  0.0,
  0.21642971,
  0.46040502,
  0.53910673,
  0.57845759,
  0.78308201,
  0.97196615,
  0.99951172,
]
```

### Error

```text
MAE  = 0.0188009664
RMSE = 0.0257373210
MAX  = 0.0393508673
```

Notably:

- eager at `x = 0.0`: `0.49975586`
- backend at `x = 0.0`: `0.53910673`

That backend value matches the LUT entry generated from the next input code, not the correct one.

## Evidence that the problem is in table generation, not table execution

The serialized TOSA emitted for the quantized sigmoid contains:

```text
tosa_table_default
aten_sigmoid_default_table_constant
```

So the quantized path is correctly being lowered to `TABLE`, not native `SIGMOID`.

Also, when comparing the generated LUT against the backend output mapping:

- the backend output matches the generated LUT entries
- the generated LUT entries do **not** match the PT2E q/dq eager reference

That indicates the bug is in **LUT construction**, not runtime application of the table.

## Likely fix

Do not use:

```python
torch.linspace(start=qmin, end=qmax, steps=256, dtype=torch.int8)
```

to generate a 256-entry int8 table domain.

Instead, generate the table input codes explicitly as integers covering the intended 256-entry TOSA int8 domain, for example:

```python
codes = torch.arange(-128, 128, dtype=torch.int16).to(torch.int8)
```

or another exact integer construction that preserves the intended code sequence without duplicates or shifts.

If the design requires some other domain than the full TOSA int8 range, it should still be built by explicit integer construction rather than integer `linspace`.

The main requirement is:

- 256 entries
- 256 distinct input codes
- exact code ordering

## Affected file

Repository path:

- `executorch/backends/arm/_passes/insert_table_ops.py`

Relevant function:

- `InsertTableOpsPass.generate_8bit_table_values()`

### Versions

Collecting environment information...
PyTorch version: 2.10.0+cpu
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Enterprise (10.0.26100 64-bit)
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.5 (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.26100-SP0
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4070 Ti
Nvidia driver version: 576.88
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Name: 13th Gen Intel(R) Core(TM) i9-13900KF
Manufacturer: GenuineIntel
Family: 207
Architecture: 9
ProcessorType: 3
DeviceID: CPU0
CurrentClockSpeed: 3000
MaxClockSpeed: 3000
L2CacheSize: 32768
L2CacheSpeed: None
Revision: None

Versions of relevant libraries:
[pip3] executorch==1.2.0.dev20260305+cpu
[pip3] numpy==2.1.3
[pip3] pytorch_tokenizers==1.1.0
[pip3] torch==2.10.0
[pip3] torchao==0.15.0
[pip3] torchvision==0.25.0
[conda] Could not collect

cc @digantdesai @freddan80 @per @zingo @oscarandersson8218 @mansnils @Sebastian-Larsson @robell

**Comment by perheld** [comment]
There should be a fix upstream soon for this.
